File: mobile/test_api.js
const http = require('http');

async function run() {
  const loginRes = await fetch('http://localhost:8080/api/auth/login', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ username: 'admin', password: 'password' })
  });
  const loginData = await loginRes.json();
  console.log('Login Response:', JSON.stringify(loginData, null, 2));
  const token = loginData.accessToken || loginData.token;
  console.log('Using Token:', token);

  const domainsRes = await fetch('http://localhost:8080/api/domains', {
    headers: { 'Authorization': 'Bearer ' + token }
  });
  const domainsData = await domainsRes.json();
  console.log('Domains Data:', JSON.stringify(domainsData, null, 2));
  
  const domains = domainsData.content || domainsData.data || (Array.isArray(domainsData) ? domainsData : []);
  if (domains.length === 0) {
    console.log('No domains found!');
    return;
  }
  const domainId = domains[0].id;

  const recordsRes = await fetch(`http://localhost:8080/api/records/domain/${domainId}?page=0&size=20`, {
    headers: { 'Authorization': 'Bearer ' + token }
  });
  const recordsData = await recordsRes.json();
  console.log('Records Response:', JSON.stringify(recordsData, null, 2));
}
run();

File: mobile/test_api.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  final loginUrl = Uri.parse('http://localhost:8080/api/auth/login');
  
  print('Logging in...');
  final loginRes = await http.post(
    loginUrl,
    headers: {'Content-Type': 'application/json'},
    body: jsonEncode({'username': 'admin', 'password': 'password'}) // use admin/admin if needed, I'll try both
  );
  
  if (loginRes.statusCode != 200) {
    print('Login failed: ${loginRes.statusCode} ${loginRes.body}');
    // let's try admin/admin
    final loginRes2 = await http.post(
      loginUrl,
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({'username': 'admin', 'password': 'admin'})
    );
    if (loginRes2.statusCode != 200) {
      print('Login with admin/admin also failed: ${loginRes2.statusCode} ${loginRes2.body}');
      return;
    }
  }

  // Assuming it succeeded, parse token
  final token = jsonDecode(loginRes.statusCode == 200 ? loginRes.body : '{"token": "dummy"}')['token'] ?? jsonDecode(loginRes.statusCode == 200 ? loginRes.body : '{"accessToken": "dummy"}')['accessToken'];
  
  print('Got token, fetching my-requests...');
  
  final reqUrl = Uri.parse('http://localhost:8080/api/approval-requests/my-requests');
  final reqRes = await http.get(reqUrl, headers: {'Authorization': 'Bearer $token'});
  
  print('Response status: ${reqRes.statusCode}');
  print('Response body: ${reqRes.body}');
}
